docker/Dockerfile

```diff
@@ -228,12 +241,8 @@ FROM dependencies-dev-platform-$TARGETARCH as dependencies-dev
 
 # Install node.js 18.x (LTS)
 # hadolint ignore=DL3008,DL4006
-RUN curl -fsSL https://deb.nodesource.com/setup_18.x | bash - && \
-    apt-get install --no-install-recommends -y nodejs && \
-    apt-get autoremove -y && \
-    apt-get clean all && \
-    rm -rf /var/lib/apt/lists/* && \
-    npm install --global markdownlint-cli@0.31.1
+COPY package.json package-lock.json /source/
+RUN npm install
```

Comment:
Something to figure out:

Our web-app `package.json` is in `nautobot/ui` and so the modules are there. But we have other developer tools we can use a top-level `package.json` for, namely `markdownlint-cli`, that having access to node modules at the top of the directory for would be nice. Until we figure out if we can have the UI source be in `nautobot/ui` but a single `package.json` at the top, this is an easier to manage divide.